src/hooks/transactions/useTransactionsStatus.tsx

```diff
@@ -0,0 +1,112 @@
+
+import useChainGetters from '../chain/useChainGetters';
+import { useSubscriptionsContext } from '../../contexts/SubscriptionsContextProvider';
+import { getChainSubscriptionsKey } from '../../util/chainsUtils';
+import { handleTransactionUpdates } from '../../util/transactionUtils';
+
+import { useSourceTarget } from '../../contexts/SourceTargetContextProvider';
+import { useApiCallsContext } from '../../contexts/ApiCallsContextProvider';
+import useLaneId from '../chain/useLaneId';
+import { useEffect } from 'react';
+import { TransactionActionCreators } from '../../actions/transactionActions';
+import isEqual from 'lodash/isEqual';
+
+export default function useTransactionsStatus(transactions: TransactionStatusType[], dispatchTransaction: Function) {
+  const { dispatchMessage } = useUpdateMessageContext();
+  const { getValuesByChain } = useChainGetters();
+
+  const subscriptions = useSubscriptionsContext();
+
+  const {
+    sourceChainDetails: { chain: currentSourceChain }
+  } = useSourceTarget();
+  const apiCalls = useApiCallsContext();
+  const laneId = useLaneId();
+
+  useEffect(() => {
+    const getTransactionStatus = async () => {
+      const updatedTransactions = await Promise.all(
+        transactions.map(async (transaction: TransactionStatusType) => {
+          if (transaction.status === TransactionStatusEnum.COMPLETED || transaction.evaluating) {
+            return transaction;
+          }
+          dispatchTransaction(
+            TransactionActionCreators.updateTransactionStatus(
+              {
+                evaluating: true
+              },
+              transaction.id
+            )
+          );
+          const { sourceChain, targetChain } = transaction;
+          const { api: targetApi } = getValuesByChain(targetChain);
+          const { sourceRole, targetRole } = getChainSubscriptionsKey({
+            currentSourceChain,
+            sourceChain
+          });
+
+          const sourceSubscriptions = subscriptions[sourceRole];
+          const targetSubscriptions = subscriptions[targetRole];
+
+          const updatedTransaction = await handleTransactionUpdates({
+            transaction,
+            sourceSubscriptions,
+            targetSubscriptions,
+            apiCalls: {
+              targetApi,
+              ...apiCalls
+            },
+            laneId,
+            dispatchMessage
+          });
+
+          if (isEqual(transaction, updatedTransaction)) {
+            dispatchTransaction(
+              TransactionActionCreators.updateTransactionStatus(
+                {
+                  evaluating: false
+                },
+                transaction.id
+              )
+            );
+          }
+
+          return updatedTransaction;
+        })
+      );
+
+      if (!isEqual(transactions, updatedTransactions)) {
+        dispatchTransaction(TransactionActionCreators.updateTransactionsStatus(updatedTransactions));
```

Comment:
How do we ensure that the update is processed in the correct order? Since this code is running async, the results might arrive in different time, hence an older update can update the newer one - since we update all transacations in one go (I'm assuming we override) - this can cause an old transaction status being displayed.

Reply:
Right. I think it was not a way to ensure the right order as you mentioned, but that was not a critical issue because the update containing the updated values would happen quite fast.

Anyway, based on your comment i redefined the evaluation process by replacing the dedicated `evaluating` state of each transaction by a global `evaluatingTransactions` state. As soon as the first evaluation starts, this state is updated and as soon as all the evaluations finishes then this variable is updated again.
With this change we make sure that the updates will come in the right order since there will not be two checks at the same time because of this check. This might decrease a little bit the performance of the check, but i prefer having the right update process.